OutOfSchool/OutOfSchool.WebApi/Services/StatisticService.cs

```diff
@@ -57,115 +62,141 @@ public async Task<IEnumerable<DirectionStatistic>> GetPopularDirections(int limi
         {
             logger.LogInformation("Getting popular categories started.");
 
-            var workshops = workshopRepository.Get<int>();
-            var applications = applicationRepository.Get<int>();
+            string cacheKey = $"GetPopularDirections_{limit}_{city}";
```

Comment:
You have same pattern twice: 
```
var cachedValue = cache.Get(cacheKey);
if (cachedValue != null) 
    return cachedValue;

var newValue = ...;
cache.Set(caheKey, newValue);

return newValue;
```

Try to introduce in `ICacheService` at least
```
T GetOrAdd(string key, T newValue)
T GetOrAdd(string key, Func<T> newValueFactory)
```
(check [ConcurrentDictionary API](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-6.0#system-collections-concurrent-concurrentdictionary-2-addorupdate(-0-system-func((-0-1))-system-func((-0-1-1)))) to get more insides about usable API)

Reply:
We created method "T GetOrAdd(string key, Func newValueFactory)"